### PR TITLE
Include epplibwrapper module

### LINKED ISSUE #431: Implementing storing and retrieving the certificate in EPP
Requires EPP test server

Work out a method of obtaining the password and certificate from CF when it is read

### PR DESCRIPTION
**Pull request by SSPJ**
## 🗣 Description ##

This PR includes a module `epplibwrapper` which wraps the epplib library (not included). This module handles setting up and tearing down connections, as well as credential management.

## 💭 Motivation and context ##

Closes #431 

## 🧪 Testing ##

1. Some additional variables in your Cloud.gov sandbox are needed. See `rotate_application_secrets.md`.

       REGISTRY_CL_ID
       REGISTRY_PASSWORD
       REGISTRY_CERT
       REGISTRY_KEY
       REGISTRY_KEY_PASSPHRASE
       REGISTRY_HOSTNAME

    To get the values for the registry test environment, contact @SSPJ .
1. Testing this in Cloud.gov will require epplib. In your requirement.txt file, add:

       git+https://github.com/cisagov/epplib.git@sspj#egg=fred-epplib
       lxml==4.9.2 ; python_version >= '2.7' and python_version not in '3.0, 3.1, 3.2, 3.3, 3.4'
       typing-extensions==4.5.0 ; python_version >= '3.7'
1. Then push your code in the usual way: `cf push getgov-you -f ops/manifests/manifest-you.yaml`
1. Log in with `cf ssh getgov-you`, `/tmp/lifecycle/shell`, `./manage.py shell`
1. Test out the connection:

       from epp.client import CLIENT as registry, commands
       req = commands.CheckHost(["ns2.okay.gov"])
       with registry.connect as wire:
          res = wire.send(req)

### CODE REVIEW

File: src/registrar/models/domain.py [hunk 1]
```diff
@@ -7,7 +7,8 @@
 from django_fsm import FSMField, transition  # type: ignore
 
 from api.views import in_domains
-from epp.mock_epp import domain_info, domain_check
+
+# from epplibwrapper import CLIENT as registry, commands
```
Comment (neilmb) [inline src/registrar/models/domain.py:11]: **question (non-blocking)**: We don't usually keep commented-out code in our files. If it makes sense here, could we also include an explanatory comment above about why it appears here?

File: src/run.sh [hunk 2]
```diff
@@ -6,4 +6,4 @@ set -o pipefail
 # Make sure that django's `collectstatic` has been run locally before pushing up to any environment,
 # so that the styles and static assets to show up correctly on any environment.
 
-gunicorn registrar.config.wsgi -t 60
+gunicorn registrar.config.wsgi -t 60 --preload
```
Comment (neilmb) [inline src/run.sh:9]: **question (non-blocking)**: It wasn't obvious to me from a quick google why we want to use this option. Could we include a comment here on why we use the `--preload` option?

**Review by neilmb — COMMENTED** [review]
Leaving comments, now I'm going to test this locally.

**Comment by neilmb** [comment]
Was successfully able to connect with the EPP server using this wrapper. :shipit:

**Review by neilmb — APPROVED** [review]

File: docs/operations/runbooks/rotate_application_secrets.md [hunk 3]
```diff
@@ -65,3 +65,41 @@ You also need to upload the `public.crt` key if recently created to the login.go
...
+openssl genpkey -out client.key \
+  -algorithm EC -pkeyopt ec_paramgen_curve:P-256 \
+  -aes-256-cbc
+openssl req -new -x509 -days 365 \
+  -key client.key -out client.crt \
+  -subj "/C=US/ST=DC/L=Washington/O=GSA/OU=18F/CN=GOV Prototype Registrar"
+
+```
+
+(Hint:
```
[the remark this answers is not in the document]
Comment (SSPJ) [reply docs/operations/runbooks/rotate_application_secrets.md]: Done.

File: src/epplibwrapper/__init__.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import logging
+from types import SimpleNamespace
+
+try:
+    from epplib import constants
+except ImportError:
```
[the remark this answers is not in the document]
Comment (SSPJ) [reply src/epplibwrapper/__init__.py:6]: Done.

File: src/epplibwrapper/client.py [hunk 5]
```diff
@@ -0,0 +1,125 @@
+import logging
```
[the remark this answers is not in the document]
Comment (SSPJ) [reply src/epplibwrapper/client.py:3]: Done.

File: src/epplibwrapper/client.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+try:
+    # Write cert and key to disk
+    CERT = Cert()
+    KEY = Key()
+except Exception as err:
+    CERT = None  # type: ignore
+    KEY = None  # type: ignore
+    logger.warning(err)
+    logger.warning(
```
[the remark this answers is not in the document]
Comment (SSPJ) [reply src/epplibwrapper/client.py:29]: TIL. Done.

File: src/epplibwrapper/client.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+        self._connect = Socket(self._client, self._login)
+
+    def _send(self, command):
+        """Helper function used by `send`."""
+        try:
+            with self._connect as wire:
+                response = wire.send(command)
+        except (ValueError, ParsingError) as err:
+            logger.debug(err)
+            logger.warning(
```
[the remark this answers is not in the document]
Comment (SSPJ) [reply src/epplibwrapper/client.py:73]: "are not enumerated" 😁

File: src/epplibwrapper/client.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        else:
+            if response.code >= 2000:
+                raise RegistryError(response.msg)
+            else:
+                return response
+
+    def send(self, command):
+        """Login, send the command, then close the connection. Tries 3 times."""
+        counter = 0  # we'll try 3 times
+        while True:
```
[the remark this answers is not in the document]
Comment (SSPJ) [reply src/epplibwrapper/client.py:109]: Good idea. Probably should have used it here, too.

File: src/registrar/models/domain.py (hunk 1 above)
Comment (SSPJ) [reply src/registrar/models/domain.py]: Was merely an example of how the wrapper will be imported. I've removed it.

File: src/run.sh (hunk 2 above)
Comment (SSPJ) [reply src/run.sh]: I can't remember, tbh. It was either helping me debug something or I needed it to fix something. I've removed `--preload`, here's hoping it was the former and not the latter!

**Pull request merged**